mezzanine/blog/management/commands/import_blogml.py

```diff
@@ -0,0 +1,127 @@
+        # empty found dict to be used mutably in post addition
+        found = {}
+        # find all categories in root, get titles and ref id"s
+        cat_elements = tree.findall(search_paths["find_categories"],
+                                    namespaces=ns)
+        # gather ref id"s on root categories into tuple
+        cat_ids = tuple((x.attrib["id"] for x in cat_elements))
+        # gather category text from title element on root categories into tuple
+        cat_title_txt = tuple(
+            (x.find("blogml:title", namespaces=ns).text for x in cat_elements))
+        # map into dictionary of key=category_id as hash,
+        # and value=category title text
+        categories_found = dict(zip(cat_ids, cat_title_txt))
+        # find all posts in root
+        posts_found = tree.findall(search_paths["find_posts"], namespaces=ns)
+        for post in posts_found:
+            """
+            By iterating on posts, extract critical information for
+            post addition.
+            - title is extracted as child of post element, text value within
+              tag
+            - content is extracted as child of post element, text value within
+              tag
+            - post-url is extracted as attribute of post element with key
+              post-url
+            - date-created is extracted as attribute of post element with name
+              date-created and converted to datetime with key date-created
+            - categories are found through dict-key comparison to parent found
+              categories, list value for key categories
+            - tags are found as grandchildren of post and children of tag with
+              attribute ref as a text description
+            """
+            found["title"] = post.find("blogml:title", namespaces=ns).text
+            found["content"] = post.find("blogml:content", namespaces=ns).text
+            found["post-url"] = post.attrib["post-url"]
+            found["date-created"] = parse(post.attrib["date-created"])
+            post_category_refs = [
+                x.attrib["ref"] for x in post.findall(
+                    search_paths["find_categories"]
+                )
+                ]
+            found["categories"] = list(
+                (categories_found[x] for x in post_category_refs))
+            found["tags"] = [x.attrib["ref"] for x in
+                             post.findall("blogml:tags/blogml:tag",
+                                          namespaces=ns)]
+            post_entered = self.add_post(title=found["title"],
+                                         content=found["content"],
+                                         old_url=found["post-url"],
+                                         pub_date=found["date-created"],
+                                         categories=found["categories"])
+            found.clear()
+            comments_found = post.findall(search_paths["find_comments"])
+            cmmnt = {}
```

Comment:
Let's use proper variable names. I see there's already a `comment` variable. How about changing them to `comment_dict` and`coment_obj`?